app/scripts/services/homePagePreference.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .service("HomePagePreference", function(){
+
+    var HOME_PAGE_PREF_KEY = "origin/home-page-pref/";
+
+    var init = function(user) {
+      HOME_PAGE_PREF_KEY += user.metadata.name;
+    };
+
+    var getHomePagePreference = function() {
+      var homePagePref = JSON.parse(localStorage.getItem(HOME_PAGE_PREF_KEY));
```

Comment:
we should have a try/catch to for JSON parse errors just in case.

Reply:
or use lodash `_.attempt()`